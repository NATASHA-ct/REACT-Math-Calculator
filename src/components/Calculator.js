import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState(0);
  const clickBtn = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  return (
    <div className="outer-container">
      <h2 className="Lets">Let&apos;s do some math!</h2>

      <div className="main-container">
        <div className="display" data-testid="display">
          <span>{state.next || state.total || 0}</span>
        </div>

        <div className="all-btns">
          {/* first row buttons */}
          <button className="operator grey" type="button" onClick={clickBtn}>
            AC
          </button>
          <button className="operator grey" type="button" onClick={clickBtn}>
            +/-
          </button>

          <button className="operator grey" type="button" onClick={clickBtn}>
            %
          </button>

          <button className="operator orange" type="button" onClick={clickBtn}>
            รท
          </button>

          {/* second row buttons */}
          <button className="num grey" type="button" onClick={clickBtn}>
            7
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            8
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            9
          </button>
          <button className="operator orange" type="button" onClick={clickBtn}>
            x
          </button>

          {/* third row buttons */}
          <button className="num grey" type="button" onClick={clickBtn}>
            4
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            5
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            6
          </button>
          <button className="operator orange" type="button" onClick={clickBtn}>
            -
          </button>

          {/* fourth row buttons */}
          <button
            className="num grey"
            data-testid="1"
            type="button"
            onClick={clickBtn}
          >
            1
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            2
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            3
          </button>
          <button className="operator orange" type="button" onClick={clickBtn}>
            +
          </button>

          {/* last row buttons */}
          <button className="num zero grey" type="button" onClick={clickBtn}>
            0
          </button>
          <button className="num grey" type="button" onClick={clickBtn}>
            .
          </button>
          <button className="operator orange" type="button" onClick={clickBtn}>
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
