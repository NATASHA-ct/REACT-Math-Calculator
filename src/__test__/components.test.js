import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from '../components/Homepage';
import Quote from "../components/quote";

describe('Rendering components Test if it satiesfies', () => {
  it('snapshot for Caculator', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot for Home', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

 it("snapshot for Quote", () => {
   const tree = renderer.create(<Quote />).toJSON();
   expect(tree).toMatchSnapshot();
 });


//   it("Renders Navbar component", () => {
//       const tree = renderer.create(
//         <MemoryRouter>
//           <Navbar />
//         </MemoryRouter>
//       ).toJSON();
//   expect(tree).toMatchSnapshot();
// });

});
