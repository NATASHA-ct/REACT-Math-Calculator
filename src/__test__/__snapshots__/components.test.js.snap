// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering components Test if it satiesfies Renders Navbar component 1`] = `
<nav
  className="navBar"
>
  <h1
    className="navHead"
  >
    Math Magicians
  </h1>
  <ul
    className="navMenu"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Rendering components Test if it satiesfies snapshot for Caculator 1`] = `
<div
  className="outer-container"
>
  <h2
    className="Lets"
  >
    Let's do some math!
  </h2>
  <div
    className="main-container"
  >
    <div
      className="display"
      data-testid="display"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="all-btns"
    >
      <button
        className="operator grey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="operator grey"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="operator grey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operator orange"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="num grey"
        data-testid="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="num zero grey"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="num grey"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Rendering components Test if it satiesfies snapshot for Home 1`] = `
<section
  className="home-section"
>
  <div
    className="home-container"
  >
    <h2
      className="home-header"
    >
      Welcome to our page!
    </h2>
    <p
      className="homepara"
    >
      A mathemagician is a mathematician who is also a magician. The term mathemagic is believed to have been introduced by Royal Vale Heath with his 1933 book Mathemagic. The name mathemagician was probably first applied to Martin Gardner, but has since been used to describe many mathematician/magicians, including Arthur T. Benjamin, Persi Diaconis, and Colm Mulcahy. Diaconis has suggested that the reason so many mathematicians are magicians is that inventing a magic trick and inventing a theorem are very similar activities. Mathemagician is a neologism, specifically a portmanteau, that combines mathematician and magician. A great number of self-working mentalism tricks rely on mathematical principles. Max Maven often utilizes this type of magic in his performance. The Mathemagician is the name of a character in the 1961 children\`s book The Phantom Tollbooth. He is the ruler of Digitopolis, the kingdom of mathematics.
    </p>
    <p
      className="homepara"
    >
      A mathemagician is a mathematician who is also a magician. The term mathemagic is believed to have been introduced by Royal Vale Heath with his 1933 book Mathemagic. The name mathemagician was probably first applied to Martin Gardner, but has since been used to describe many mathematician/magicians, including Arthur T. Benjamin, Persi Diaconis, and Colm Mulcahy. Diaconis has suggested that the reason so many mathematicians are magicians is that inventing a magic trick and inventing a theorem are very similar activities. Mathemagician is a neologism, specifically a portmanteau, that combines mathematician and magician. A great number of self-working mentalism tricks rely on mathematical principles. Max Maven often utilizes this type of magic in his performance. The Mathemagician is the name of a character in the 1961 children\`s book The Phantom Tollbooth. He is the ruler of Digitopolis, the kingdom of mathematics.
    </p>
  </div>
</section>
`;

exports[`Rendering components Test if it satiesfies snapshot for Quote 1`] = `
<div
  className="quoteCont"
>
  <h3
    className="quoteHead"
  >
     
    “Mathematics is a hard thing to love. It has the unfortunate habit, like a rude dog, of turning its most unfavorable side towards you when you first make contact with it.”
    <br />
    (David Whiteland),  mental calculator, 10 Jun 1944
  </h3>
</div>
`;
